python/pyspark/sql/functions.py

```diff
@@ -1705,10 +1705,12 @@ def unhex(col):
 @ignore_unicode_prefix
 @since(1.5)
 def length(col):
-    """Calculates the length of a string or binary expression.
+    """Computes the character length of a given string or number of bytes or a binary string.
+    The length of character strings include the trailing spaces. The length of binary strings
```

Comment:
as a side note, why is it calling out trailing spaces? what about leading spaces? isn't all spaces factored into the character length?

Reply:
I ask because I want to understand this better to see if we should update R https://github.com/apache/spark/blob/master/R/pkg/R/functions.R#L1029

Reply:
The reason is `LEN` in MS SQL Server excludes trailing blanks. : )

Yeah. This PR also can updates it in R side too.